bin/single_target

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env ruby
+
+require "aws-sdk-elasticloadbalancingv2"
+
+TARGETS_FILE_PATH = '/tmp/deploy_tools_single_tgs'
+
+function = ARGV[0]
+$host_id = ARGV[1]
+
+fail "Please specify AWS_PROFILE" if ENV["AWS_PROFILE"].nil?
+fail "Usage: single_target <deregister || register> <host_id>" if (function.nil? or $host_id.nil?)
+
+REGION = "us-west-2".freeze
+class AWSClient
+  class << self
+    def elbv2
+      @elbv2 ||= Aws::ElasticLoadBalancingV2::Client.new(region: REGION)
+    end
+  end
+end
+
+def find_host_tgs ()
+  all_tgs = AWSClient.elbv2.describe_target_groups.target_groups
+  host_tgs = []
+
+  all_tgs.each do |tg|
+    targets = AWSClient.elbv2.describe_target_health(target_group_arn: tg.target_group_arn)
+                    .target_health_descriptions.map(&:target)
+    target_ports = targets.map(&:port).uniq
+
+    if targets.map(&:id).include?($host_id)
+      host_tgs.push ({
+        :target_group_arn => tg.target_group_arn,
+        :target_group_name => tg.target_group_name,
+        :target_current_port => target_ports[0]
+      })
+    end
+  end
+
+  fail "Target not found in any target group!" if host_tgs.length == 0
+  File.write(TARGETS_FILE_PATH, host_tgs.join("\n"))
```

Comment:
writing non-serialized data to file? I mean if you want to do, better to serialize it

Reply:
It turns out I need to (de-)serialize data even if not writing to file.